### DISCUSSION TITLE
About Emulating Materials with Imaginary Parts Causes Total Efficiency Overflow

### DISCUSSION THREAD

**Discussion by xungu1** [kc-ml2/meent#87] [General]
Hello, I recently used meent to simulate materials with imaginary parts. For meent, it seems that as long as the material has an imaginary part, the total efficiency will exceed 1, even if it is only one layer. I conducted experiments on zhaonat-rcwa-yonghakim, Maxim and MatRCWA-master on githud. In zhaonat 's program, it seems that energy is conserved only when the real part of the dielectric constant is positive and the imaginary part is negative, while in MAxim and MatRCWA it does not seem to affect. The energy conservation in the literature I have found should be α + t + r = 1, and α is the absorption rate, but no specific parameters are specified. The files in the following compression package are calculated using the latest meent. The material text file is the refractive index at a wavelength of 400nm-750nm.
[result.zip](https://github.com/user-attachments/files/18251451/result.zip)

**Comment by yonghakim** [comment]
Hi, the sum of total efficiency is one only if all refractive indices of the materials are real. When any of them is complex, it can be less than one or more than one depending on the sign of the imaginary part.
You may refer the last slide of this: https://empossible.net/wp-content/uploads/2019/08/Lecture-7a-RCWA-Formulation.pdf

**Comment by xungu1** [comment]
Yes, I have also noticed on the page about empossible. What makes me curious is whether the absorption rate α can be calculated ? I have noticed that some references mention that the TM polarization may lead to overflow when simulating metals, but there will also be overflow under TE polarization, but some people say that metals are energy-conserving at certain wavelengths.

**Comment by yonghakim** [comment]
I'm not aware of any methods predicting the loss or gain.

Meent uses Li's inverse rule to enhance the convergence. I remember some paper mentioned this may be bad for metal simulation.
I recommend to see Reticolo manual and its code. This gives special option for metal.
